### PR TITLE
Us041 scheduled state

### PR DESCRIPTION
**Pull request by madeye-matt**
When mandatory events are added, the collection exercise state becomes SCHEDULED

### CODE REVIEW

**Comment by gemmai95** [comment]
[US041: Create and View 'Scheduled' State of a CE [M]](https://trello.com/c/Quhfml2t/35-us041-create-and-view-scheduled-state-of-a-ce-m)

File: src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/EventServiceImpl.java [hunk 1]
```diff
@@ -211,4 +216,24 @@ public void clearEventMessageSent(UUID eventId) throws CTPException {
...
+    @Override
+    public boolean isScheduled(UUID collexUuid) throws CTPException {
+        Map<String, Event> events = getEvents(collexUuid).stream().collect(
+                Collectors.toMap(Event::getTag, Function.identity())
+        );
+
+        return Arrays.stream(Tag.values())
+                .map(t -> events.get(t.name()))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList())
```
Comment (benjefferies) [inline src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/EventServiceImpl.java]: We could use .count() here

File: src/main/java/uk/gov/ons/ctp/response/collection/exercise/state/CollectionExerciseStateTransitionManagerFactory.java [hunk 2]
```diff
@@ -53,16 +53,36 @@ public CollectionExerciseStateTransitionManagerFactory() {
 
     Map<CollectionExerciseState, Map<CollectionExerciseEvent, CollectionExerciseState>> transitions = new HashMap<>();
 
-    // INIT
+    // INIT/CREATED
```
Comment (benjefferies) [inline src/main/java/uk/gov/ons/ctp/response/collection/exercise/state/CollectionExerciseStateTransitionManagerFactory.java]: Could probably be `// CREATED`

**Comment by benjefferies** [comment]
Super-seeded by https://github.com/ONSdigital/rm-collection-exercise-service/pull/30/files

**Pull request closed without merging**